ereuse_devicehub/resources/device/models.py

```diff
@@ -256,6 +256,31 @@ def working(self):
                                         key=attrgetter('type'))  # last test of each type
         return self._warning_actions(current_tests)
 
+    def merge_device(self, with_device):
+        """Merge the current device with `with_device` by
+        adding all `with_device` actions under the current device.
+
+        This operation is highly costly as it forces refreshing
+        many models in session.
+        """
+        assert isinstance(with_device, int)
+        w_device = Device.query.filter_by(id=with_device).one()
+        for action in w_device.actions:
+
+            if action.parent:
+                action.parent = self
+            else:
+                action.device_id = self.id
+
+        for component in w_device.components:
+            if component.parent:
+                component.parent = self
```

Comment:
the components can not just be reassigned, but they have to be derived from the latest snapshot action.

e.g. merge two devices
- base: device A, components C1, C2, C3, actions A1 (Snapshot, registered 10 January 2020), A2 (Ready)
- merged into device A: device B, components C4, C5, actions A3 (Snapshot, registered 10 May 2020), A4 (Snapshot, registered 30 May 2020)

the resulting device A **does not have all** components C1, C2, C3, C4, C5.

the components are determined by the latest snapshot action. In the example, the latest snapshot action is A4. therefore, the components of A4 determine the device's components. 

similary, `hid`, `model`, `manufacturer` and `chassis` are also determined by latest snapshot action. A4 in the example